### DISCUSSION TITLE
reverse-proxy timeout how to set?

### DISCUSSION THREAD

**Discussion by duyanming** [dotnet/yarp#2183] [Q&A] [answered]
I'm using.net 7

I want to know how to set the reverse-proxy timeout period

**Comment by Tratcher** [comment, marked as answer]
See ActivityTimeout:
https://github.com/microsoft/reverse-proxy/blob/5cad42d60a514f2b93fc47968d4405c6d06d39f2/docs/docfx/articles/http-client-config.md?plain=1#L95
https://github.com/microsoft/reverse-proxy/blob/5cad42d60a514f2b93fc47968d4405c6d06d39f2/docs/docfx/articles/config-files.md?plain=1#L218
https://github.com/microsoft/reverse-proxy/blob/5cad42d60a514f2b93fc47968d4405c6d06d39f2/docs/docfx/articles/direct-forwarding.md?plain=1#L59

**Comment by duyanming** [comment]
The ActivityTimeout explanation seems to say that this is the connection idle recycle time, what I want is Yarp request target timeout, thank you
I don't know if there's something wrong with my understanding

**Reply by duyanming** [reply]
In addition, I would like to know how to obtain the event after the service health monitoring is offline.

I am looking forward to your reply

thank you

**Reply by Tratcher** [reply]
ActivityTimeout applies to active requests too, not just idle connections. If a request does not make progress within the given time period it will be aborted.

**Reply by duyanming** [reply]
thank you
